Description: Allow an empty static alias file
 An empty static alias file specified via, e.g.,
    table aliases file:/etc/aliases
 will cause smtpd to abort with
    invalid use of table "aliases" as ALIAS parameter.
 This is because it doesn't get assigned the T_DYNAMIC flag by virtue of using
 the static backend. Moreover, since it contains no lines, the backend is unable
 to divine whether the file should be treated as a T_HASH or a T_LIST, and so
 neither of these flags. Thus, when smtpd checks to see that the « aliases »
 table satisfies T_DYNAMIC|T_HASH (neither of which are satisfied by the empty
 aliases file) at startup, the check fails and smtpd aborts.
Author: Ryan Kavanagh <rak@debian.org>
Origin: upstream
Applied-Upstream: https://github.com/poolpOrg/OpenSMTPD/commit/3ed3ceb4
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: opensmtpd/smtpd/table_static.c
===================================================================
--- opensmtpd.orig/smtpd/table_static.c	2013-08-15 08:05:28.543759950 -0400
+++ opensmtpd/smtpd/table_static.c	2013-08-15 08:05:28.543759950 -0400
@@ -137,6 +137,10 @@
 		else
 			goto end;
 	}
+	/* Accept empty alias files; treat them as hashes */
+	if (t->t_type == T_NONE && t->t_backend->services & K_ALIAS)
+	    t->t_type = T_HASH;
+
 	ret = 1;
 end:
 	free(lbuf);
